src/DistributedWriteAheadLog/tests/gtest_record_sender.cpp

```diff
@@ -91,14 +91,57 @@ TEST(CheckRecordSerializationDeserialization, Serder)
 
         for (size_t i = 0; i < col_got.column->size(); ++i)
         {
-            if (colname == "id")
+            if (col_name == "id")
                 EXPECT_EQ(col_expected.column->get64(i), col_got.column->get64(i));
-            else if (colname == "cpu" || colname == "_time")
+            else if (col_name == "cpu" || col_name == "_time")
                 EXPECT_EQ(col_expected.column->getFloat64(i), col_got.column->getFloat64(i));
-            else if (colname == "raw")
+            else if (col_name == "raw")
                 EXPECT_EQ(col_expected.column->getDataAt(i), col_got.column->getDataAt(i));
             else
                 ASSERT_TRUE(false);
         }
     }
 }
+
+TEST(CheckRecordSerializationDeserialization, Sender)
+{
+    auto r = createRecord();
+    ByteVector data{Record::write(*r, false)};
+    auto rr = Record::read(reinterpret_cast<char *>(data.data()), data.size());
+    auto cols = std::vector<String>{"id", "cpu", "raw", "_time"};
+
+    checkRecord(cols, *r, *rr);
+}
+
+TEST(CheckRecordSerializationDeserialization, Compression)
+{
+    auto r = createRecord();
+    ByteVector data{Record::write(*r, true)};
+    auto rr = Record::read(reinterpret_cast<char *>(data.data()), data.size());
+    auto cols = std::vector<String>{"id", "cpu", "raw", "_time"};
+
+    checkRecord(cols, *r, *rr);
+}
+
+TEST(CheckRecordSerializationDeserialization, Benchmark)
+{
+    auto r = createRecord();
+    /// 100000 write/read
+    auto write_read_loop = [&](bool compressed)
+    {
+        for(auto i=0;i<100000;i++)
```

Comment:
The code is not correctly formatted. What about the new code. Still the same speed ?

Reply:
Yes, the speed is almost the same.